src/redux/books/booksSlice.js

```diff
@@ -1,39 +1,83 @@
-import { createSlice } from '@reduxjs/toolkit';
+import { createSlice, createAsyncThunk } from '@reduxjs/toolkit';
+import axios from 'axios';
 
-const initialState = [
-  {
-    item_id: 'item1',
-    title: 'The Great Gatsby',
-    author: 'John Smith',
-    category: 'Fiction',
-  },
-  {
-    item_id: 'item2',
-    title: 'Anna Karenina',
-    author: 'Leo Tolstoy',
-    category: 'Fiction',
-  },
-  {
-    item_id: 'item3',
-    title: 'The Selfish Gene',
-    author: 'Richard Dawkins',
-    category: 'Nonfiction',
-  },
-];
+const initialState = {
+  books: [],
+  isLoading: false,
+};
+
+const getBooks = createAsyncThunk('books/getBooks', async (thunkAPI) => {
+  try {
+    const response = await axios.get('https://us-central1-bookstore-api-e63c8.cloudfunctions.net/bookstoreApi/apps/CmHBlH9icvMuSf2R7xNo/books');
+    return response.data;
+  } catch (error) {
+    return thunkAPI.rejectWithValue('Failed, Try again!');
+  }
+});
+
+const addBook = createAsyncThunk('books/addBook', async (bookInfo, thunkAPI) => {
+  try {
+    const response = await axios.post('https://us-central1-bookstore-api-e63c8.cloudfunctions.net/bookstoreApi/apps/CmHBlH9icvMuSf2R7xNo/books', bookInfo);
+    return response.data;
+  } catch (error) {
+    return thunkAPI.rejectWithValue('Failed, Try again!');
+  }
+});
```

Comment:
- When we add a book kindly make sure to the application is rerendering well, with the right list of books, when you add a book.

Here is the behavior of your app when we add a book:
[Screencast from 08-04-2023 10:25:53.webm](https://user-images.githubusercontent.com/97834160/230711774-fba9a1c5-51a3-44d0-ab51-4884acf12316.webm)

Reply:
Done